### PR TITLE
User Permission Frontend

### PR DESCRIPTION
**Pull request by pbardea**
Addresses the rest of #72 

When adding a user you can add them as a collaborator or as a watcher...

Currently can't change the permissions of existing users.

Also I just added another button. Let me know if you feel strongly about changing that.

### CODE REVIEW

File: backend/routes/users.js [hunk 1]
```diff
@@ -50,10 +64,21 @@ router.put('/:tripId/email/:userEmail', (req, res) => {
     }
     const newRelation = {
       userId: user.id,
-      tripId: req.params.tripId
+      tripId: req.params.tripId,
+      readOnly: JSON.parse(readOnlyPerm),
```
Comment (aneedalie) [inline backend/routes/users.js]: You already do a parse above, do you need to do it again?

File: frontend/src/components/Collaborators.vue [hunk 2]
```diff
@@ -26,7 +26,8 @@
         </div>
 
         <input type="email" v-model="email" placeholder="name@domain.com"></input>
-        <button class="btn btn-success" v-on:click="addCollaborator()">Add</button>
+        <button class="btn btn-success" v-on:click="addCollaborator(false)">Add as Collaborator</button>
```
Comment (aneedalie) [inline frontend/src/components/Collaborators.vue]: I find this quite misleading that the collaborator one takes in a `false` while the Watcher is a `true`.

Comment (aneedalie) [reply frontend/src/components/Collaborators.vue]: I was going to have a button, where `v-on:click="readOnly = !readOnly"` and then in the addCollaborator function i just read that. But that does require more UI work.

Comment (pbardea) [reply frontend/src/components/Collaborators.vue]: i can try doing that later. you mean a toggle button right?

**Pull request merged**